### DISCUSSION TITLE
Migrate from Gateway or Filesystem Mode

### DISCUSSION THREAD

**Discussion by whitepiratebaku** [minio/minio#18330] [Q&A] [answered]
Hello, we are stuck at RELEASE.2022-10-15T19-57-03Z and because of breaking issues we can't upgrade our MinIO, I see there is a migration [guide](https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html) for the upgrade but unfortunately we do not have extra space for mirroring, and we have large amount of data. Like above 17TB.  We wonder is there anyway to upgrade minio while using existing data and make it quickly? because doing "mc mv localfolder minio_alias/bucket/" would take days with current speed of disks. Any workarounds?

**Comment by klauspost** [comment, marked as answer]
No.
